### DISCUSSION TITLE
PropTypes with observableRequiresReaction check

### DISCUSSION THREAD

**Discussion by hymm** [mobxjs/mobx#2621] [Q&A] [answered]
I tried enabling observableRequiresReaction, but it's erroring on the react proptype checks. i.e.

```js
function Component({ value }) {
  return <div>{value}</div>;
}

Component.propTypes = {
  // the next line is causing a observableRequiresReaction warning, wrapping it in an action removes the warning.
  value: PropTypes.number.isRequired, 
}
```

Is there a better way of dealing with this other than wrapping the propType checker in an action?

**Comment by danielkcz** [comment]
Well, you should wrap the component to `observer`, that check is kinda correct here. If you are wrapping it later on export, that won't work imo.

**Reply by hymm** [reply]
Here's a codesandbox with the issue: https://codesandbox.io/s/lucid-merkle-5cw3z?file=/index.js.  My earlier example wasn't the best.

**Comment by mweststrate** [comment, marked as answer]
This is a known issue, propTypes are checked outside the render. You can
either use the propTypes provided by mobx-react, or ignore the warnings, or
use a type checker instead of propTypes :)
